src/document.rs

```diff
@@ -192,32 +191,17 @@ fn extract_value_single_or_list_for_type(
     }
 }
 
-fn value_to_object(val: &JsonValue, py: Python<'_>) -> PyObject {
-    match val {
-        JsonValue::Null => py.None(),
-        JsonValue::Bool(b) => b.to_object(py),
-        JsonValue::Number(n) => match n {
-            n if n.is_i64() => n.as_i64().to_object(py),
-            n if n.is_u64() => n.as_u64().to_object(py),
-            n if n.is_f64() => n.as_f64().to_object(py),
-            _ => panic!("number too large"),
-        },
-        JsonValue::String(s) => s.to_object(py),
-        JsonValue::Array(v) => {
-            let inner: Vec<_> =
-                v.iter().map(|x| value_to_object(x, py)).collect();
-            inner.to_object(py)
-        }
-        JsonValue::Object(m) => {
-            let inner: HashMap<_, _> =
-                m.iter().map(|(k, v)| (k, value_to_object(v, py))).collect();
-            inner.to_object(py)
-        }
+fn object_to_py(py: Python, obj: &BTreeMap<String, Value>) -> PyResult<PyObject> {
+    let dict = PyDict::new(py);
+    for (k, v) in obj.iter() {
+        dict.set_item(k, value_to_py(py, v)?)?;
     }
+    Ok(dict.into())
 }
 
 fn value_to_py(py: Python, value: &Value) -> PyResult<PyObject> {
     Ok(match value {
+        Value::Null => py.None(),
```

Comment:
`Null` and `Array` have been added into the value types, and `JsonObject` changed to just `Object`.  There are no tests for the new types yet.  I won't add tests for these extra fields in this PR, that can come in a future PR before we do a release with the new version.